subworkflows/local/binning_refinement.nf

```diff
@@ -0,0 +1,124 @@
+                                        def meta_new = meta.clone()
+                                        [ meta_new, assembly ]
+                                }
+
+    ch_bins_for_fastatocontig2bin = RENAME_PREDASTOOL(bins).renamed_bins
+                                        .branch {
+                                            metabat2: it[0]['binner'] == 'MetaBAT2'
+                                            maxbin2:  it[0]['binner'] == 'MaxBin2'
+                                        }
+
+    // Generate DASTool auxilary files
+    DASTOOL_FASTATOCONTIG2BIN_METABAT2 ( ch_bins_for_fastatocontig2bin.metabat2, "fa")
+    DASTOOL_FASTATOCONTIG2BIN_MAXBIN2 ( ch_bins_for_fastatocontig2bin.maxbin2, "fasta")
+
+    // Run DASTOOL
+    ch_fastatocontig2bin_for_dastool = Channel.empty()
+    ch_fastatocontig2bin_for_dastool = ch_fastatocontig2bin_for_dastool
+                                    .mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.fastatocontig2bin)
+                                    .mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.fastatocontig2bin)
+                                    .map {
+                                        meta, fastatocontig2bin ->
+                                            def meta_new = meta.clone()
+                                            meta_new.remove('binner')
+                                            [ meta_new, fastatocontig2bin ]
+                                    }
+                                    .groupTuple(by: 0)
+
+    ch_input_for_dastool = ch_contigs_for_dastool.join(ch_fastatocontig2bin_for_dastool, by: 0)
+
+    ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.versions.first())
+    ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.versions.first())
+
+    // Run DAStool
+    DASTOOL_DASTOOL(ch_input_for_dastool, [], [])
+
+    ch_dastool_bins_newmeta = DASTOOL_DASTOOL.out.bins.transpose()
+        .map {
+            meta, bin ->
+                def meta_new = meta.clone()
+
+                meta_new['binner'] = bin.name.split("-")[1]
+                [ meta_new, bin ]
+            }
+        .groupTuple()
+
+    ch_versions = ch_versions.mix(DASTOOL_DASTOOL.out.versions.first())
+
+    ch_input_for_renamedastool = DASTOOL_DASTOOL.out.bins
+        .map {
+            meta, bins ->
+                def meta_new = meta.clone()
+                meta_new['binner'] = 'DASTool'
+                [ meta_new, bins ]
+            }
+
+    RENAME_POSTDASTOOL ( ch_input_for_renamedastool )
+
+    // We have to strip the meta to be able to combine with the original
+    // depths file to run MAG_DEPTH
+    ch_input_for_magdepth = ch_dastool_bins_newmeta
+        .mix( RENAME_POSTDASTOOL.out.refined_unbins )
+        .map {
+                meta, refinedbins ->
+                def meta_new = [ 'id': meta.id, 'group': meta.group, 'single_end': meta.single_end, 'assembler': meta.assembler ]
```

Comment:
Is there a reason you didn't use `meta_new.remove('binner')` here as above? If not then this might help readability

```suggestion
                def meta_new = meta.clone()
                meta_new.remove('binner')
```